server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/DoubleRareTerms.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.aggregations.bucket.terms;
+
+import org.apache.lucene.util.NumericUtils;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.util.BloomFilter;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.BucketOrder;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.InternalAggregations;
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Result of the RareTerms aggregation when the field is some kind of decimal number like a float, double, or distance.
+ */
+public class DoubleRareTerms extends InternalMappedRareTerms<DoubleRareTerms, DoubleTerms.Bucket> {
```

Comment:
I'd love for us to not implement this one since doubles and floats are likely to have an extremely long tail and therefore make using this aggregation on them untenable